services/caption_video.py

```diff
@@ -37,35 +37,55 @@
     'Korean Bold': '/usr/share/fonts/truetype/custom/Korean-Bold.ttf'
 }
 
-def process_captioning(file_url, caption_srt, options, job_id):
+def process_captioning(file_url, caption_srt, caption_type, options, job_id):
     """Process video captioning using FFmpeg."""
     try:
         logger.info(f"Job {job_id}: Starting download of file from {file_url}")
         video_path = download_file(file_url, STORAGE_PATH)
         logger.info(f"Job {job_id}: File downloaded to {video_path}")
 
-        srt_path = os.path.join(STORAGE_PATH, f"{job_id}.srt")
-
+        subtitle_extension = '.' + caption_type
+        srt_path = os.path.join(STORAGE_PATH, f"{job_id}{subtitle_extension}")
+        options = convert_array_to_collection(options)
+        caption_style = ""
+
+        if caption_type == 'ass':
+            style_string = generate_style_line(options)
+            caption_style = f"""
+[Script Info]
+Title: Highlight Current Word
+ScriptType: v4.00+
+[V4+ Styles]
+Format: Name, Fontname, Fontsize, PrimaryColour, OutlineColour, BackColour, Bold, Italic, Underline, StrikeOut, ScaleX, ScaleY, Spacing, Angle, BorderStyle, Outline, Shadow, Alignment, MarginL, MarginR, MarginV, Encoding
+{style_string}
+[Events]
+Format: Layer, Start, End, Style, Name, MarginL, MarginR, MarginV, Effect, Text
+"""
+        
         if caption_srt.startswith("https"):
             # Download the file if caption_srt is a URL
             logger.info(f"Job {job_id}: Downloading caption file from {caption_srt}")
             response = requests.get(caption_srt)
             response.raise_for_status()  # Raise an exception for bad status codes
-            
-            with open(srt_path, 'wb') as srt_file:
-                srt_file.write(response.content)
+            if caption_type in ['srt','vtt']:
+                with open(srt_path, 'wb') as srt_file:
+                    srt_file.write(response.content)
+            else:
```

Comment:
The caption type should be fixed for .ass. It looks now that if it's anything other than SRT or VTT, it will still run.